substrate/pkg/controller/substrate/cluster/addons/karpenter.go

```diff
@@ -39,8 +39,17 @@ kind: Provisioner
 metadata:
   name: default
 spec:
+  requirements:
+  - key: kit.k8s.sh/app
```

Comment:
Why do we need both?

Reply:
Currently we have two nodeselectors on the API server and etcd pods to one to identify which control plane they belong to and another one to distinguish among themselves. This we will need to revisit at some point as these node selectors will change.